tests/subsys/net/lib/nrf_cloud/nrf_cloud_test/src/nrf_cloud_test.c

```diff
@@ -0,0 +1,609 @@
+		"return should be non-zero when transport routine init failed");
+}
+
+/* Verify that nrf_cloud_init function will fail
+ * when full modem FOTA update fails to initialize.
+ * Ignored when full modem FOTA update is disabled
+ */
+void test_init_fota_failed(void)
+{
+#if defined(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE)
+	struct dfu_target_fmfu_fdev fmfu_dev_info = {
+		.dev = NULL,
+		.offset = 0L,
+		.size = 0U,
+	};
+
+	struct nrf_cloud_init_param params = {
+		.event_handler = event_handler,
+		.client_id = NULL,
+		.fmfu_dev_inf = &fmfu_dev_info,
+	};
+
+	/* Expect an error return value for nrf_cloud_fota_fmfu_dev_set */
+	nrf_cloud_fota_fmfu_dev_set_fake.custom_fake = nrf_cloud_fota_fmfu_dev_set_failed;
+
+	/* Expect other fakes to run successfully */
+	nct_init_fake.custom_fake = nct_init_success;
+	nfsm_init_fake.custom_fake = nfsm_init_success;
+	nrf_cloud_codec_init_fake.custom_fake = nrf_cloud_codec_init_success;
+
+	TEST_ASSERT_EQUAL_MESSAGE(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	int ret = nrf_cloud_init(&params);
+
+	TEST_ASSERT_NOT_EQUAL_MESSAGE(0, ret,
+		"return should be non-zero when FOTA init failed");
+#else
+	TEST_IGNORE();
+#endif
+}
+
+/* TEST NFSM_SET_CURRENT_STATE_AND_NOTIFY */
+
+void test_set_current_state(void)
+{
+	TEST_ASSERT_EQUAL_MESSAGE(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	nfsm_set_current_state_and_notify(STATE_TOTAL, NULL);
+
+	TEST_ASSERT_EQUAL_MESSAGE(STATE_TOTAL, nfsm_get_current_state(),
+		"nrf_cloud lib should be in STATE_TOTAL after changing");
+}
+
+/* TEST NRF_CLOUD_UNINIT */
+
+void test_uninit_fota_active(void)
+{
+#if defined(CONFIG_NRF_CLOUD_FOTA)
+	TEST_ASSERT_EQUAL_MESSAGE(STATE_IDLE, nfsm_get_current_state(),
+		"nrf_cloud lib should be in the idle state (uninitialized) at the start of test");
+
+	struct nrf_cloud_init_param params = {
+		.event_handler = event_handler,
+		.client_id = NULL,
+	};
+
+	nct_init_fake.custom_fake = nct_init_success;
+	nfsm_init_fake.custom_fake = nfsm_init_success;
+	nrf_cloud_codec_init_fake.custom_fake = nrf_cloud_codec_init_success;
+#if defined(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE)
+	nrf_cloud_fota_fmfu_dev_set_fake.custom_fake = nrf_cloud_fota_fmfu_dev_set_success;
+#endif
+	nrf_cloud_init(&params);
+
+	/* Expect nrf_cloud_fota_uninit fake returns busy */
+	nrf_cloud_fota_uninit_fake.custom_fake = nrf_cloud_fota_uninit_busy;
+	nct_disconnect_fake.custom_fake = nct_disconnect_success;
+
+	int ret = nrf_cloud_uninit();
+
+	TEST_ASSERT_EQUAL_MESSAGE(-EBUSY, ret,
+		"return should be -EBUSY when a FOTA job is active");
+#else
+	TEST_IGNORE();
+#endif
+}
+
+/* This test is ignored when the connection poll thread is active
+ * since it will triggered the transport disconnect event and give
+ * the uninit_disconnect semaphore which fails the test eventually
+ */
```

Comment:
I would place this comment inside the function, perhaps right before TEST_IGNORE